src/explore-education-statistics-frontend/src/modules/table-tool/components/TableHeadersForm.tsx

```diff
@@ -30,42 +36,183 @@ const TableHeadersForm = (props: Props) => {
+                                      draggableSnapshot.isDragging,
+                                  },
+                                )}
+                                ref={draggableProvided.innerRef}
+                              >
+                                <FormFieldSortableList<FormValues>
+                                  name={`${group}[${index}]`}
+                                  id={`sort-${group}-${index}`}
+                                  legend={`${groupLegend} ${index + 1}`}
+                                  legendSize="s"
+                                />
+                              </div>
+                            )}
+                          </Draggable>
+                        ))}
+                        {droppableProvided.placeholder}
+                      </div>
+                    </FormFieldset>
+                  </div>
+                )}
+              </Droppable>
+            );
+          };
+
           return (
             <Form>
-              <div className="govuk-grid-row">
-                <div className="govuk-grid-column-one-quarter">
-                  <FormFieldSortableList<FormValues>
-                    name="rowGroups"
-                    id="sort-rowGroups"
-                    legend="Row groups"
-                  />
-                </div>
-                <div className="govuk-grid-column-one-quarter">
-                  <FormFieldSortableList<FormValues>
-                    name="rows"
-                    id="sort-rows"
-                    legend="Rows"
-                  />
-                </div>
-                <div className="govuk-grid-column-one-quarter">
-                  <FormFieldSortableList<FormValues>
-                    name="columnGroups"
-                    id="sort-columnGroups"
-                    legend="Column groups"
-                  />
+              <DragDropContext
+                onDragEnd={result => {
+                  if (!result.destination) {
+                    return;
+                  }
+
+                  const { source, destination } = result;
+
+                  const destinationId = destination.droppableId as
+                    | 'columnGroups'
+                    | 'rowGroups';
+
+                  const sourceId = source.droppableId as
+                    | 'columnGroups'
+                    | 'rowGroups';
+
+                  const destinationGroup = form.values[destinationId];
+
+                  if (destinationId === sourceId) {
+                    form.setFieldTouched(destinationId);
+                    form.setFieldValue(
+                      destinationId,
+                      reorder(
+                        destinationGroup,
+                        source.index,
+                        destination.index,
+                      ),
+                    );
+                  } else {
+                    const sourceClone = Array.from(form.values[sourceId]);
```

Comment:
Is "Array.from()" the preferred syntax for creating arrays over "[...form.values[sourceId]]" ?

Reply:
This was mostly a lift from a `react-beautiful-dnd` (drag and drop) example :smile: I would say we should prefer to use the spread syntax `[...form.values[sourceId]]` because it looks nicer, but it doesn't really make a difference in most cases (`Array.from` is actually a little different in how it works).